wgpu-core/src/device/mod.rs

```diff
@@ -998,13 +998,34 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         let (format, view_kind, range) = match desc {
             Some(desc) => {
                 let kind = conv::map_texture_view_dimension(desc.dimension);
+                let required_level_count =
+                    desc.base_mip_level + desc.level_count.and_then(|l| Some(l.get())).unwrap_or(1);
```

Comment:
we could write this simpler as `desc.level_count.map_or(1, |count| count.get())`